streamlit_app.py

```diff
@@ -0,0 +1,448 @@
+import streamlit as st
+
+import sys
+sys.path.append('src')
+import deck52
+import pprint
+import time
+import json
+import uuid
+import shelve
+
+import numpy as np
+
+import human
+import bots
+import sample
+
+from bidding import bidding
+from nn.models import Models
+from deck52 import decode_card
+from bidding.binary import DealData
+from objects import CardResp, Card
+
+# test dds load, encounter libboost_thread.so.1.71.0 not found
+#from ddsolver import dds
+
+def print(r):
+  st.write(r)
+
+#bids = st.text_input("Standard american biding convention", "1D 2D")
+#st.write(f"The meaning of {bids} is")
+
+
+def random_deal():
+    deal_str = deck52.random_deal()
+    auction_str = deck52.random_dealer_vuln()
+
+    return deal_str, auction_str
+
+
+class AsyncBotBid(bots.BotBid):
+    def async_bid(self, auction):
+        return self.bid(auction)
+
+class AsyncBotLead(bots.BotLead):
+    def async_lead(self, auction):
+        return self.lead(auction)
+
+class AsyncCardPlayer(bots.CardPlayer):
+    def async_play_card(self, trick_i, leader_i, current_trick52, players_states):
+        return self.play_card(trick_i, leader_i, current_trick52, players_states)
+
+
+class Driver:
```

Comment:
this `Driver` class is almost identical with the one in `game.py`
since stopping after the bidding is an interesting feature that could be useful in general, i think it would be nice to extend the existing `Driver` instead of redefining it. that way this functionality could be used in other places too, not just streamlit.